aten/src/ATen/native/TypeProperties.cpp

```diff
@@ -169,7 +169,7 @@ ScalarType result_type(ITensorListRef tensors) {
 ScalarType result_type(const Tensor &tensor, const Tensor &other) {
   // NOLINTNEXTLINE(performance-move-const-arg)
   std::vector<Tensor> tensors{std::move(tensor), std::move(other)};
```

Comment:
Actually, this one is wrong (look at the args to the function they are const ref. Should pass by value.

Reply:
The two tensors want to be moved into a vector, then the vector is used to create the result. In theory we can alway move a temporary object like this vector.  In this case the two moves have no effect as explained by you, but we can move the vector to avoid an additional copy. Another way is to change the signature to ScalarType result_type(Tensor tensor, Tensor other) and then construct a std::initialization_list by {std::move(tensor),std::move(other)} without creating the vector.

Reply:
At the risk of changing  the API, I prefer the latter approach.